pilot/pkg/model/context.go

```diff
@@ -616,6 +617,32 @@ func (m NodeMetadata) ProxyConfigOrDefault(def *meshconfig.ProxyConfig) *meshcon
 	return def
 }
 
+// IdleTimeoutOrDefault returns the idle timeout of the proxy in a durationpb.Duration format if applicable.
+// A nil return implies the using of default idle timeout.
+func (m NodeMetadata) IdleTimeoutOrDefault() *durationpb.Duration {
+	if m.IdleTimeout == "" {
+		// do nothing and use default if no idle timeout duration is specified
+		return nil
+	}
+
+	idleTimeout, err := time.ParseDuration(m.IdleTimeout)
+	if err != nil {
+		log.Warnf("Failed to parse the specified idle timeout for the proxy: %v, use default value", err)
+		return nil
+	}
+
+	if idleTimeout < 0 {
+		log.Warnf("Configured with an invalid negative idle timeout %v for the proxy, use default value", idleTimeout)
+		return nil
+	}
+
+	if idleTimeout == 0 {
+		log.Warn("Disabling the proxy idle timeout, which has a highly likelihood of yielding connection leaks " +
```

Comment:
this is not the best place to log this, it will spam the logs a lot. If we need this info it belongs in an analyzer probably

Reply:
or maybe in the webhook as a warning, then user is most likely to see it

Reply:
I kind of having this concern too. Just two questions:
1. Will explicitly setting idle timeout as 0s be a normal case?
2. Are you suggesting to not log at all in this function, or only for `timeout == 0`?
Thanks!

Reply:
> Will explicitly setting idle timeout as 0s be a normal case?

No, defaults are normal case. But that doesn't mean we should log 1000s of times if they do set it. The problem with logging here is its called many many times - potentionally n^2 for the number of connections.

We should only log it one time - when they apply the config. Which can be done by adding some warning in install or webhooks or analyzer or similar?